Dockerfile

```diff
@@ -25,16 +11,14 @@ COPY setup.cfg .
 COPY setup.py .
 COPY README.md .
 RUN pip install -e .
-
-COPY dev-requirements.txt dev-requirements.txt
-RUN pip install -r dev-requirements.txt
+# TODO(epwalsh): In PyTorch 1.7, dataclasses is an unconditional dependency, when it should
+# only be a conditional dependency for Python < 3.7.
+# This has been fixed on PyTorch master branch, so we should be able to
+# remove this check with the next PyTorch release.
+RUN pip uninstall -y dataclasses
 
 COPY allennlp_server/ allennlp_server/
 
-RUN pip install -e .
-
-LABEL maintainer="allennlp-contact@allenai.org"
+RUN pip install --no-deps -e .
```

Comment:
Why `--no-deps`?

Reply:
It saves some time since we already install the deps in a previous step. It also ensures that `dataclasses` doesn't get installed again (we explicitly uninstall it in a previous step).